### ISSUE TITLE
Implement APRS BASE91 Comment Telemetry

### ISSUE THREAD

**Issue by kb1lqc** [FaradayRF/Faraday-Software#266]
## Summary
8 Bit telemetry is, well, antiquated. This issue ticket will implement BASE91 comment telemetry into the `faraday-aprs` package. The result will be a full 12 bit value for each telemetry field.

## Problem Explanation
Some developers added on the ability for APRS-IS and websites such as APRS.fi to support BASE91 encoded telemetry in comments. This allows up to 16 bit accuracy for each telemetry field.

#### Software 
>Please include the branch or commit if known regarding what software was used if applicable

#### Hardware
>Let us know about what hardware you are using if applicable. This includes:
>- Faraday radio serial numbers
>- Computer operating system

# Supporting Information

- https://aprs.fi/doc/guide/aprsfi-telemetry.html
- http://he.fi/doc/aprs-base91-comment-telemetry.txt

**Comment by kb1lqc** [comment]
Installing `base91` from pip: https://pypi.python.org/pypi/base91/1.0.1

**Comment by kb1lqc** [comment]
Oh sweet Jesus... this stumped me for a few minutes

```
3.4 The telemetry sequence counter and telemetry channels are encoded using
the Base91 encoding, as it is presently used in Compressed APRS position
packets, the altitude extension, and the DAO extension. Two bytes are
transmitted for the sequence counter and each of the channels, giving over
13 bits of resolution (values 0 to 8280). Please note that APRS uses a
different definition of Base91 than the internet standard Base91 - see
APRS101.PDF for details.
```

> Please note that APRS uses a different definition of Base91 than the internet standard Base91 - see APRS101.PDF for details.

🤔

**Comment by kb1lqc** [comment]
Well even more reason to move to `aprslib`, implements base91 APRS flavor https://github.com/rossengeorgiev/aprs-python/blob/master/aprslib/base91.py

![standards](https://user-images.githubusercontent.com/331540/30040705-c2a786ca-9197-11e7-8e37-8b7551315c4c.png)

**Comment by kb1lqc** [comment]
Alright I got it. It turns out that `aprslib` (or [aprs-python](https://github.com/rossengeorgiev/aprs-python)) has a BASE91 encoder/decoder. 

I completely missed that you could specify width with `aprslib` BASE91... woops. I thought there was an error

**Comment by kb1lqc** [comment]
Ran code overnight to get 12 bit ADC data into APRS-IS!

![image](https://user-images.githubusercontent.com/331540/30062408-eb38670e-91ff-11e7-8783-47457c466986.png)

![image](https://user-images.githubusercontent.com/331540/30062416-f22afe14-91ff-11e7-8443-1578e813ced4.png)

Still had a problem of APRS-IS disconnecting overnight but progress.

**Comment by kb1lqc** [comment]
It worked well most of today and the data looks really nice.

![image](https://user-images.githubusercontent.com/331540/30101944-f32a4aec-92a2-11e7-9956-cbeeae4eb4fc.png)

I lost connection in the evening. So there still needs to be some TLC on connecting to APRS-IS. Also, I should add IO Bits to the BASE91 data before closing this one.

**Comment by kb1lqc** [comment]
Added BASE91 IO data to the comment telemetry transmitted with each position report in branch aprsTelem 0a57a3b

![image](https://user-images.githubusercontent.com/331540/30150529-98768944-9360-11e7-9f53-9ba18683872c.png)

**Comment by kb1lqc** [comment]
WHOA @aprs.fi on twitter helped me get the network adapter working right in Virtualbox. I had to use a "bridged" network adapter setting after power cycling the debian server! I then could connect to the IP Address but would get booted and wouldn't show up on the aprsc status as a client! Restartedthe server and changed my APRS-IS name to `KB1LQC-16" and it works!
![image](https://user-images.githubusercontent.com/331540/30151417-30f1cb18-9364-11e7-8a92-e54495b6cb75.png)

Sending these through my locally hosted APRS-IS `aprsc` server! @kb1lqd @el-iso @reillyeon Now I will be able to test connection issues with ease :)

![image](https://user-images.githubusercontent.com/331540/30151466-694c2c38-9364-11e7-9fe9-c376d65acfc6.png)

**Comment by kb1lqc** [comment]
Still going strong! I hope aprsc running locally helps smooth over network disconnects. If this ends up working over long durations then I can definitely start using local aprsc to mimic network loss and make my code much more reliable! @hdkmike @kb1lqd 

https://aprs.fi/telemetry/a/KB1LQC-1?range=day

![image](https://user-images.githubusercontent.com/331540/30167979-f030449a-939c-11e7-837c-5ba12a8716ba.png)

**Comment by kb1lqc** [comment]
Also, going on 7 hours client connection. Looking good!
![image](https://user-images.githubusercontent.com/331540/30168479-611e2252-939e-11e7-8971-c8c646db7b36.png)

This will help #199, #190, and #119 though I should probably move over to one of those tickets when I actually start coding to resolve this issue. This ticket is just for BASE91 telemetry support which appears to be working!

**Comment by kb1lqc** [comment]
OK got kicked off:

![image](https://user-images.githubusercontent.com/331540/30196149-440059d6-9413-11e7-824f-04104598f5dc.png)

![image](https://user-images.githubusercontent.com/331540/30196170-5d00ed06-9413-11e7-9fca-537969533da4.png)

my local aprsc server remained connected to APRS-IS so the problem must be with `faraday-aprs` not handling even local network blips. 
![image](https://user-images.githubusercontent.com/331540/30196218-c54dd6da-9413-11e7-836c-94f4ad2485c4.png)

OK I should move this to a different ticket as this is the BASE91 ticket!

**Comment by kb1lqc** [comment]
Closed with #269 merge. Woot!
